controller/api/destination/watcher/workload_watcher.go

```diff
@@ -145,6 +155,16 @@ func (ww *WorkloadWatcher) Unsubscribe(ip string, port Port, listener WorkloadUp
 	if len(wp.listeners) == 0 {
 		delete(ww.publishers, IPPort{wp.addr.IP, wp.addr.Port})
 	}
+
+	ww.updateSubscriberCount()
+}
+
+func (ww *WorkloadWatcher) updateSubscriberCount() {
+	totalSubscribers := 0
+	for _, wp := range ww.publishers {
+		totalSubscribers += len(wp.listeners)
+	}
```

Comment:
How many publishers are there going to be in a loaded system? Is it really necessary to iterate over all on every subscribe/unsubscribe? Is it possible to keep a running total (i.e. an Atomic) that keeps the tally without requiring iteration?